pkg/install/generate.go

```diff
@@ -197,82 +177,74 @@ func GenerateInstallerJob(
-				// Inlining a script to be run, we cannot assume a script to be in older images, nor that older images will output
-				// a sleep-seconds.txt file. If one is written, we will sleep that number of seconds. This allows exponential backoff
-				// for failing installs.
-				fmt.Sprintf(
-					"/usr/bin/hiveutil install-manager --work-dir /output --log-level debug --region %s %s %s; installer_result=$?; if [ -f /output/sleep-seconds.txt ]; then sleep_seconds=$(cat /output/sleep-seconds.txt); echo \"sleeping for $sleep_seconds seconds until next retry\"; sleep $sleep_seconds; fi; exit $installer_result",
-					cd.Spec.Platform.AWS.Region,
-					cd.Namespace,
-					cd.Name),
+			Command:         []string{"/usr/bin/hiveutil"},
+			Args: []string{"install-manager",
+				"--work-dir", "/output",
+				"--log-level", "debug",
+				"--region", cd.Spec.Platform.AWS.Region,
+				cd.Namespace, provisionName,
 			},
 			VolumeMounts: volumeMounts,
 		},
 	}
 
-	restartPolicy := corev1.RestartPolicyOnFailure
-	if tryOnce {
-		restartPolicy = corev1.RestartPolicyNever
-	}
-
-	podSpec := corev1.PodSpec{
+	return &corev1.PodSpec{
 		DNSPolicy:          corev1.DNSClusterFirst,
-		RestartPolicy:      restartPolicy,
+		RestartPolicy:      corev1.RestartPolicyNever,
 		Containers:         containers,
 		Volumes:            volumes,
 		ServiceAccountName: serviceAccountName,
 		ImagePullSecrets: []corev1.LocalObjectReference{
 			cd.Spec.PullSecret,
 		},
-	}
+	}, nil
+}
 
-	completions := int32(1)
-	backoffLimit := int32(123456) // effectively limitless
-	if tryOnce {
-		backoffLimit = int32(0)
-	}
+// GenerateInstallerJob creates a job to install an OpenShift cluster
+// given a ClusterDeployment and an installer image.
+func GenerateInstallerJob(provision *hivev1.ClusterProvision) (*batchv1.Job, error) {
 
-	labels := map[string]string{
-		InstallJobLabel:            "true",
-		ClusterDeploymentNameLabel: cd.Name,
-	}
-	if cd.Labels != nil {
-		typeStr, ok := cd.Labels[hivev1.HiveClusterTypeLabel]
-		if ok {
-			labels[hivev1.HiveClusterTypeLabel] = typeStr
-		}
-	}
+	pLog := log.WithFields(log.Fields{
+		"clusterProvision": provision.Name,
+		"namespace":        provision.Namespace,
+	})
+
+	pLog.Debug("generating installer job")
 
-	annotations := map[string]string{
-		clusterDeploymentGenerationAnnotation: strconv.FormatInt(cd.Generation, 10),
+	labels := provision.Labels
+	if labels == nil {
+		labels = map[string]string{}
 	}
+	labels[InstallJobLabel] = "true"
+	labels[ClusterDeploymentNameLabel] = provision.Spec.ClusterDeployment.Name
 
 	job := &batchv1.Job{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:        GetInstallJobName(cd),
-			Namespace:   cd.Namespace,
-			Annotations: annotations,
-			Labels:      labels,
+			Name:      GetInstallJobName(provision),
+			Namespace: provision.Namespace,
+			Labels:    labels,
 		},
 		Spec: batchv1.JobSpec{
-			Completions:  &completions,
-			BackoffLimit: &backoffLimit,
+			Completions: pointer.Int32Ptr(1),
```

Comment:
Any benefits to using the Job still or should we switch to run-once Pods?

Reply:
The main benefit for using jobs instead of pods is that it is more efficient to watch jobs than to watch all the pods in the cluster as there are far fewer jobs than pods. It is also somewhat easier to ascertain whether a job was successful then whether a pod was successful. I have been torn on whether to scrap jobs in favor of pods, too. Can we put a pin in that and bring it up for discussion and as a candidate for a later PR?

Reply:
SGTM